source/common/http/filter/gzip_filter.cc

```diff
@@ -0,0 +1,159 @@
+#include "common/http/filter/gzip_filter.h"
+
+#include <regex>
+
+#include "common/common/macros.h"
+
+namespace Envoy {
+namespace Http {
+
+static const std::regex& acceptEncodingRegex() {
+  CONSTRUCT_ON_FIRST_USE(std::regex, "(?!.*gzip;\\s*q=0(,|$))(?=(.*gzip)|(^\\*$))",
+                         std::regex::optimize);
+}
+
+GzipFilterConfig::GzipFilterConfig(const envoy::api::v2::filter::http::Gzip& gzip)
+    : compression_level_(compressionLevelEnum(gzip.compression_level())),
+      compression_strategy_(compressionStrategyEnum(gzip.compression_strategy())),
+      content_length_(static_cast<uint64_t>(gzip.content_length().value())),
+      memory_level_(static_cast<uint64_t>(gzip.memory_level().value())),
+      etag_(gzip.disable_on_etag().value()),
+      last_modified_(gzip.disable_on_last_modified().value()) {
+
+  for (const auto& value : gzip.cache_control()) {
+    cache_control_values_.insert(value);
+  }
+
+  for (const auto& value : gzip.content_type()) {
+    content_type_values_.insert(value);
+  }
+}
+
+ZlibCompressionLevelEnum
+GzipFilterConfig::compressionLevelEnum(const GzipV2CompressionLevelEnum& compression_level) {
+  if (compression_level == GzipV2CompressionLevelEnum::Gzip_CompressionLevel_Enum_BEST) {
+    return ZlibCompressionLevelEnum::Best;
+  }
+  if (compression_level == GzipV2CompressionLevelEnum::Gzip_CompressionLevel_Enum_SPEED) {
+    return ZlibCompressionLevelEnum::Speed;
+  }
+  return ZlibCompressionLevelEnum::Standard;
+}
+
+ZlibCompressionStrategyEnum GzipFilterConfig::compressionStrategyEnum(
+    const GzipV2CompressionStrategyEnum& compression_strategy) {
+  if (compression_strategy == GzipV2CompressionStrategyEnum::Gzip_CompressionStrategy_RLE) {
+    return ZlibCompressionStrategyEnum::Rle;
+  }
+  if (compression_strategy == GzipV2CompressionStrategyEnum::Gzip_CompressionStrategy_FILTERED) {
+    return ZlibCompressionStrategyEnum::Filtered;
+  }
+  if (compression_strategy == GzipV2CompressionStrategyEnum::Gzip_CompressionStrategy_HUFFMAN) {
+    return ZlibCompressionStrategyEnum::Huffman;
+  }
+  return ZlibCompressionStrategyEnum::Standard;
+}
+
+const uint64_t GzipFilter::WINDOW_BITS{15 | 16};
```

Comment:
Can you add a comment about where these values come from?

Reply:
I added this to get rid of the magic number, but good point about the comment.